sdk/trace/trace_test.go

```diff
@@ -470,39 +469,70 @@ func TestSamplerAttributesLocalChildSpan(t *testing.T) {
 }
 
 func TestSetSpanAttributesOverLimit(t *testing.T) {
-	te := NewTestExporter()
-	tp := NewTracerProvider(WithSpanLimits(SpanLimits{AttributeCountLimit: 2}), WithSyncer(te), WithResource(resource.Empty()))
+	// The tracing specification states:
+	//
+	//   Setting an attribute with the same key as an existing attribute
+	//   SHOULD overwrite the existing attribute's value.
 
-	span := startSpan(tp, "SpanAttributesOverLimit")
-	span.SetAttributes(
+	te := NewTestExporter()
+	tp := NewTracerProvider(
+		WithSpanLimits(SpanLimits{AttributeCountLimit: 2}),
+		WithSyncer(te),
+		WithResource(resource.Empty()),
+	)
+	attrs := []attribute.KeyValue{
 		attribute.Bool("key1", true),
 		attribute.String("key2", "value2"),
 		attribute.Bool("key1", false), // Replace key1.
-		attribute.Int64("key4", 4),    // Remove key2 and add key4
-	)
+		attribute.Int64("key4", 4),    // Dropped
+	}
+
+	span := startSpan(tp, "SpanAttributesOverLimit")
+	span.SetAttributes(attrs...)
 	got, err := endSpan(te, span)
 	if err != nil {
 		t.Fatal(err)
 	}
+	assert.Contains(t, got.attributes, attrs[1])
+	assert.Contains(t, got.attributes, attrs[2])
```

Comment:
Can we also assert the DroppedAttributes() == 1?
```suggestion
	assert.Contains(t, got.attributes, attrs[1])
	assert.Contains(t, got.attributes, attrs[2])
	asster.Equal(t, got.droppedAttributes, 1)
```